app/src/main/java/com/kickstarter/ui/adapters/DiscoveryPagerAdapter.java

```diff
@@ -1,25 +1,24 @@
 package com.kickstarter.ui.adapters;
 
 import android.support.annotation.NonNull;
-import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentManager;
 import android.support.v4.app.FragmentPagerAdapter;
 import android.view.ViewGroup;
 
 import com.kickstarter.libs.utils.DiscoveryUtils;
 import com.kickstarter.models.Category;
 import com.kickstarter.services.DiscoveryParams;
-import com.kickstarter.ui.ArgumentsKey;
 import com.kickstarter.ui.fragments.DiscoveryFragment;
 
+import java.util.HashMap;
 import java.util.List;
-
-import rx.Observable;
+import java.util.Map;
 
 public final class DiscoveryPagerAdapter extends FragmentPagerAdapter {
   private final Delegate delegate;
   private final FragmentManager fragmentManager;
   private List<String> pageTitles;
+  private Map<Integer, DiscoveryFragment> fragmentMap;
```

Comment:
Main question I have here is whether there are any issues with holding onto a reference to these fragments – can it potentially cause any issues with GC? I couldn't see a clear explanation on how this works from a scan of the docs, so would just hope that we have confidence that this would not cause issues with destroying fragments.